src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -3364,6 +3375,60 @@ void writeBigDecimal(BigDecimal bigDecimalVal,
         System.arraycopy(valueBytes, 2, bytes, 0, valueBytes.length - 2);
         writeBytes(bytes);
     }
+    
+    /**
+     * Append a big decimal inside sql_variant in the TDS stream.
+     * 
+     * @param bigDecimalVal
+     *            the big decimal data value
+     * @param srcJdbcType
+     *            the source JDBCType
+     */
+    void writeSqlVariantInternalBigDecimal(BigDecimal bigDecimalVal,
+            int srcJdbcType) throws SQLServerException {
+        /*
+         * Length including sign byte One 1-byte unsigned integer that represents the sign of the decimal value (0 => Negative, 1 => positive) One
+         * 16-byte signed integer that represents the decimal value multiplied by 10^scale. In sql_variant, we send the bigdecimal with precision 38,
+         * therefore we use 16 bytes for the maximum size of this integer.
+         */
+
+        boolean isNegative = (bigDecimalVal.signum() < 0);
+        BigInteger bi = bigDecimalVal.unscaledValue();
+        if (isNegative)
+            bi = bi.negate();
+        int bLength;
+        bLength = BYTES16;
+
+        writeByte((byte) (isNegative ? 0 : 1));
+
+        // Get the bytes of the BigInteger value. It is in reverse order, with
+        // most significant byte in 0-th element. We need to reverse it first before sending over TDS.
+        byte[] unscaledBytes = bi.toByteArray();
+
+        if (unscaledBytes.length > bLength) {
+            // If precession of input is greater than maximum allowed (p><= 38) throw Exception
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_valueOutOfRange"));
+            Object[] msgArgs = {JDBCType.of(srcJdbcType)};
+            throw new SQLServerException(form.format(msgArgs), SQLState.DATA_EXCEPTION_LENGTH_MISMATCH, DriverError.NOT_SET, null);
+        }
+
+        // Byte array to hold all the reversed and padding bytes.
+        byte[] bytes = new byte[bLength];
+
+        // We need to fill up the rest of the array with zeros, as unscaledBytes may have less bytes
+        // than the required size for TDS.
+        int remaining = bLength - unscaledBytes.length;
+
+        // Reverse the bytes.
+        int i, j;
+        for (i = 0, j = unscaledBytes.length - 1; i < unscaledBytes.length;)
+            bytes[i++] = unscaledBytes[j--];
```

Comment:
same for this. This was an existing code, I moved part of it to new method as it was partially common code. For this one, since it was existing I will keep it as is if you agree?